src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -667,19 +685,42 @@ public void onInitialize(Run run) {
 
     }
 
-
-    @SafeVarargs
-    private static void assertPollingBaselines(List<WorkflowRun.SCMCheckout> checkouts, Matcher<Object>... indexedMatchers) {
-        assertThat("Number of checkouts should match number of matchers", checkouts.size(), equalTo(indexedMatchers.length));
-        for (int i = 0; i < checkouts.size(); i++) {
-            assertThat("Unexpected baseline for checkout at index " + i, checkouts.get(i).pollingBaseline, indexedMatchers[i]);
+    @Issue("JENKINS-73835")
+    @Test public void logRotationOnlyProcessesCompletedBuilds() throws Throwable {
+        logging.record(LogRotator.class, Level.FINER);
+        var p = r.createProject(WorkflowJob.class);
+        p.setDefinition(new CpsFlowDefinition(
+                "echo params.FOO; semaphore 'wait'", true));
+        p.addProperty(new ParametersDefinitionProperty(List.of(new StringParameterDefinition("FOO"))));
+        // Keep 0 builds, i.e. delete all builds immediately.
```

Comment:
I am keeping 0 builds rather than 1 build to simplify test assertions. Mainly I am trying to avoid complicated issues involving the exact timing of calls to `RunList.size` and `RunList.sublist` in `LogRotator.perform` compared to other builds completing, which can cause builds other than the one with the largest build number to be preserved by `LogRotator`.